src/commander.ts

```diff
@@ -421,50 +421,45 @@ export class Commander {
         }
         const configuration = vscode.workspace.getConfiguration('latex-workshop')
         if (!configuration.get('bind.enter.key')) {
-            return editor.edit(() =>
-                vscode.commands.executeCommand('type', { source: 'keyboard', text: '\n' })
-            )
+            return vscode.commands.executeCommand('type', { source: 'keyboard', text: '\n' })
         }
         if (modifiers === 'alt') {
             return vscode.commands.executeCommand('editor.action.insertLineAfter')
         }
 
+        // Test if every cursor is at the end of a line starting with \item
+        const allCursorsOnItem = editor.selections.every((selection: vscode.Selection) => {
+                const cursorPos = selection.active
+                const line = editor.document.lineAt(cursorPos.line)
+                return /^\s*\\item/.test(line.text) && (line.text.substring(cursorPos.character).trim().length === 0)
+        })
+        if (!allCursorsOnItem) {
+            return vscode.commands.executeCommand('type', { source: 'keyboard', text: '\n' })
+        }
 
         void editor.edit(editBuilder => {
```

Comment:
Instead of `void`, it would be better to return the value that `editor.edit` returns.